packages/main/src/FileUploader.ts

```diff
@@ -317,6 +360,33 @@ class FileUploader extends UI5Element implements IFormInputElement {
 		}, "");
 	}
 
+	/**
+	 * Checks whether all files are below `maxFileSize` (if set),
+	 * and fires a `fileSizeExceeded` event if any file exceeds it.
+	 *
+	 * @returns true if all files are below `maxFileSize`
+	 * @private
+	 */
+	_validateFileSize(files: FileList | null): boolean {
+		if (!this.maxFileSize || !files) {
+			return true;
+		}
+
+		const convertBytesToMegabytes = (bytes: number) => (bytes / 1024) / 1024;
```

Comment:
This function should be extracted to module level, f.e. at the start of the file before the class definition.